### ISSUE TITLE
Address Book

### ISSUE THREAD

**Issue by fubuloubu** [ApeWorX/ape#306]
### Overview
It is sometime really handy to have a set of "trusted" addresses that I'd want to reference by alias. It would also be really handy to have an address book to use when I am doing some work in a script but don't want to completely dox someone's address.

### Specification
```bash
$ ape accounts save 0xAbCd...1234 my-friend
[INFO] Account `0xAbCd...1234` saved as `my-friend`
```

```py
from ape import accounts

alice = accounts.load("me")
bob = accounts.load("my-friend")
alice.transfer(bob, "1 ether")
```

**Comment by fubuloubu** [comment]
Could be a use case to add a second account container to `ape_accounts` that are just basic `ape.Address` classes

**Comment by fubuloubu** [comment]
idea: might even be cool to have different address books you can switch between for different projects :thinking:

**Comment by unparalleled-js** [comment]
Should we also allow this to be done via conifg?

```yaml
contacts:
  my_friend: "0x1122334455677E444496A5555A258Aff9333B1002"
```

```python
from ape import accounts

my_friend = accounts.load("my_friend")
```

**Comment by fubuloubu** [comment]
> Should we also allow this to be done via conifg?

Yes, but I've been thinking more and more lately that it might make sense to have a "global config" in the data folder, for things like your network preferences. This address book concept would probably fit nicely too. Basically anything that is a user preference and not a project one.

The "global config" would be the same format as the local one, but be retreived first, with the local applied over it, in this order:

config defaults > global config > local config

**Comment by unparalleled-js** [comment]
Oh, nice, yes that makes sense. I will create a ticket for that.

**Comment by unparalleled-js** [comment]
Would it makes sense for the global config to be in the `.ape/` application directory or would it make more sense to have it just be in the user's `$HOME` directory (like a global `.gitignore`)?

I am leaning on the data directory, like you suggest, but I am a little concerned of that UX as users don't typically interact with application data folders

**Comment by fubuloubu** [comment]
> Would it makes sense for the global config to be in the `.ape/` application directory or would it make more sense to have it just be in the user's `$HOME` directory (like a global `.gitignore`)?
> 
> I am leaning on the data directory, like you suggest, but I am a little concerned of that UX as users don't typically interact with application data folders

Could be `.ape/config.yaml` yeah, but also... could make a whole config CLI to ease the setting of popular user-preference types of config (e.g. what provider plugins to install, favorite network, etc.)

**Comment by unparalleled-js** [comment]
Like

`ape config add-plugin`
`ape config add-account`
`ape config set-network`

Or even

`ape config set-editor vi` (or `code` or `emacs` or whatever editor binary command you use).
`ape config edit` and that opens up the `config.yaml` file and allows you to edit it.

might also need something like `ape --config --show-path` to show the path to the config file controlling the setting

**Comment by fubuloubu** [comment]
hmm, a lot in there. maybe:

```
$ ape config set-default ethereum rinkeby infura
$ ape config --list
```

**Comment by fubuloubu** [comment]
Decided to make this into a plugin: https://github.com/ApeWorX/ape-addressbook
